seed/seed20.go

```diff
@@ -0,0 +1,348 @@
+			revRefs = append(revRefs, ref)
+		}
+		return nil
+	}
+
+	batch, err := loadAssertions(assertsDir, checkAssertion)
+	if err != nil {
+		return err
+	}
+
+	refs, err := readAsserts(batch, filepath.Join(s.systemDir, "model"))
+	if err != nil {
+		return fmt.Errorf("cannot read model assertion: %v", err)
+	}
+	if len(refs) != 1 || refs[0].Type != asserts.ModelType {
+		return fmt.Errorf("system model assertion file must contain exactly the model assertion")
+	}
+	modelRef := refs[0]
+
+	// this also verifies the consistency of all of them
+	if err := commitTo(batch); err != nil {
+		return err
+	}
+
+	find := func(ref *asserts.Ref) (asserts.Assertion, error) {
+		a, err := ref.Resolve(db.Find)
+		if err != nil {
+			return nil, fmt.Errorf("internal error: cannot find just accepted assertion %v: %v", ref, err)
+		}
+		return a, nil
+	}
+
+	a, err := find(modelRef)
+	if err != nil {
+		return err
+	}
+	modelAssertion := a.(*asserts.Model)
+
+	snapDeclsByName := make(map[string]*asserts.SnapDeclaration, len(declRefs))
+	snapDeclsByID := make(map[string]*asserts.SnapDeclaration, len(declRefs))
+
+	for _, declRef := range declRefs {
+		a, err := find(declRef)
+		if err != nil {
+			return err
+		}
+		snapDecl := a.(*asserts.SnapDeclaration)
+		snapDeclsByID[snapDecl.SnapID()] = snapDecl
+		if snapDecl1 := snapDeclsByName[snapDecl.SnapName()]; snapDecl1 != nil {
+			return fmt.Errorf("cannot have multiple snap-declarations for the same snap-name: %s", snapDecl.SnapName())
+		}
+		snapDeclsByName[snapDecl.SnapName()] = snapDecl
+	}
+
+	snapRevsByID := make(map[string]*asserts.SnapRevision, len(revRefs))
+
+	for _, revRef := range revRefs {
+		a, err := find(revRef)
+		if err != nil {
+			return err
+		}
+		snapRevision := a.(*asserts.SnapRevision)
+		snapRevision1 := snapRevsByID[snapRevision.SnapID()]
+		if snapRevision1 != nil {
+			if snapRevision1.SnapRevision() != snapRevision.SnapRevision() {
+				return fmt.Errorf("cannot have multiple snap-revisions for the same snap-id: %s", snapRevision1.SnapID())
+			}
+		} else {
+			snapRevsByID[snapRevision.SnapID()] = snapRevision
+		}
+	}
+
+	// remember db for later use
+	s.db = db
+	// remember
+	s.model = modelAssertion
+	s.snapDeclsByID = snapDeclsByID
+	s.snapDeclsByName = snapDeclsByName
+	s.snapRevsByID = snapRevsByID
+
+	return nil
+}
+
+func (s *seed20) Model() (*asserts.Model, error) {
+	if s.model == nil {
+		return nil, fmt.Errorf("internal error: model assertion unset")
+	}
+	return s.model, nil
+}
+
+func (s *seed20) UsesSnapdSnap() bool {
+	return true
+}
+
+func (s *seed20) loadAuxInfos() error {
+	auxInfoFn := filepath.Join(s.systemDir, "snaps", "aux-info.json")
+	if !osutil.FileExists(auxInfoFn) {
+		// missing
+		return nil
+	}
+
+	f, err := os.Open(auxInfoFn)
+	if err != nil {
+		return err
+	}
+	defer f.Close()
+	dec := json.NewDecoder(f)
+	if err := dec.Decode(&s.auxInfos); err != nil {
+		return fmt.Errorf("cannot decode aux-info.json: %v", err)
+	}
```

Comment:
do we want to check `dec.More()`?

Reply:
to error or to accept more maps?  I have do admit I'm too fussed either way